src/cpp/rtps/writer/StatefulWriter.cpp

```diff
@@ -493,7 +493,7 @@ bool StatefulWriter::intraprocess_heartbeat(
     bool returned_value = false;
 
     std::lock_guard<RecursiveTimedMutex> guardW(mp_mutex);
-    RTPSReader* reader = RTPSDomainImpl::find_local_reader(reader_proxy->guid());
+    RTPSReader* reader = reader_proxy->local_reader();
```

Comment:
I think this should be left untouched.

There is a corner case where the proxy could still have a pointer to an already deleted reader.